### ISSUE TITLE
Add better support for renamed NAACCR IDs

### ISSUE THREAD

**Issue by depryf** [imsweb/naaccr-xml#153]
The current version already contains special support for the renamed NAACCR IDs in version 18: it contains a method `NaaccrXmlDictionaryUtils.getRenamedLongNaaccr18Ids()` that provides a static mapping of the old vs new IDs.

I thought this would be enough to fix applications that need to read XML data and support old and new IDs for a while. But that is still not enough. The problem is that the library itself detects bad NAACCR IDs and reports an error, and when that happens, the value is not copied into the read data structure. There is an option to process bad IDs, which would allow the application to have the value and translate the old IDs into the new ones, but that means legit bad IDs would also be processed. Not great at all.

I think I am going to add another option (which will be OFF by default so it doesn't affect anything); it will tell the library to automatically translate the old IDs into the new ones (that will only happen for NAACCR 18, and only for those specific renamed ones).

**Comment by depryf** [comment]
I added the option (NaaccrOptions.setTranslateRenamedItemIds) and tested it and it works great. I also added that option to the standalone tool, which allows one to read an XML data file with the old IDs and re-create the same file with the new IDs ("fixing" the file).

Unfortunately the testing also revealed a case that is still hard to handle: if the application using this library defines a user-defined dictionary that has long IDs (longer than 32), those IDs have to be shorten, but there is no way for the application to supporting reading the long version of those IDs for a while. The library needs a way to allow the application to register a map of IDs to be translated. That's the same idea as the translated standard items, but for the non-standard ones...

**Comment by depryf** [comment]
I still need to finish some testing, but here is how it's supposed to work.

To translate the standard items that got renamed in N18, one needs to call
```java
option.setTranslateRenamedStandardItemIds(true);
```
The default for that option is false.

To translate non-standard items, one has to call 
```java
options.setItemIdsToTranslate(map);
```
And provide a map with the old ID/new ID. Technically standard item IDs can be provided in that map but I don't see any advantages in doing that.

**Comment by depryf** [comment]
This will be released with version 6.1
